app/cdap/components/Connections/Browser/GenericBrowser/BrowserTable.tsx

```diff
@@ -28,18 +28,42 @@ import DescriptionIcon from '@material-ui/icons/Description';
 import Heading, { HeadingTypes } from 'components/Heading';
 import LoadingSVG from 'components/LoadingSVG';
 import { format } from 'services/DataFormatter';
+import IconSVG from 'components/IconSVG';
 
-const ICON_MAP = {
-  directory: <FolderIcon />,
-  file: <DescriptionIcon />,
-};
+function getIcon(entity, classes) {
+  const type = entity.type.toLowerCase();
+
+  switch (type) {
+    case 'bucket':
```

Comment:
Nit: I'd prefer to have a comment calling out the non-exhaustive switch

Reply:
Noted. I won't make the change here since it's a cherry pick but I'll keep that in mind in the future.